libs/api_common/map_descriptors.cpp

```diff
@@ -88,21 +89,30 @@ cache_map_file_descriptor_with_handle(
     uint32_t key_size,
     uint32_t value_size,
     uint32_t max_entries,
+    uint32_t inner_map_fd,
     int fd,
     uintptr_t handle,
     size_t section_offset)
 {
-    _map_file_descriptors.emplace_back(handle, fd, type, key_size, value_size, max_entries, 0, section_offset);
+    _map_file_descriptors.emplace_back(
+        handle, fd, type, key_size, value_size, max_entries, inner_map_fd, section_offset);
 }
 
 int
 cache_map_handle(
-    uint64_t handle, uint32_t type, uint32_t key_size, uint32_t value_size, uint32_t max_entries, size_t section_offset)
+    uint64_t handle,
+    uint32_t type,
+    uint32_t key_size,
+    uint32_t value_size,
+    uint32_t max_entries,
+    uint32_t inner_map_fd,
+    size_t section_offset)
 {
-    // TODO: Replace this with the CRT helper to create FD from handle once we
+    // TODO(issue #287): Replace this with the CRT helper to create FD from handle once we
     // have real handles.
```

Comment:
Actually this comment is not valid now. cache_map_handle() is called from create_map_internal() and is always expected to create a mock fd and never an actual fd.

Reply:
removed comment